realtime_ai_character/llm/openai_llm.py

```diff
@@ -33,16 +37,38 @@ def __init__(self, model):
                 streaming=True
             )
         self.db = get_chroma()
+        self.search_agent = None
+        if os.getenv('SERPER_API_KEY'):
+            llm = OpenAI(temperature=0)
+            tools = load_tools(["google-serper"], llm=llm)
+            self.search_agent = initialize_agent(
+                tools, llm, agent=AgentType.ZERO_SHOT_REACT_DESCRIPTION
+            )
 
     async def achat(self,
                     history: List[BaseMessage],
                     user_input: str,
                     user_input_template: str,
                     callback: AsyncCallbackTextHandler,
                     audioCallback: AsyncCallbackAudioHandler,
-                    character: Character) -> str:
+                    character: Character,
+                    useSearch: bool=False) -> str:
         # 1. Generate context
         context = self._generate_context(user_input, character)
+        # Get search result if enabled
+        if useSearch:
+            if self.search_agent is None:
+                logger.warning('Search is not enabled, please set SERPER_API_KEY to enable it.')
+            else:
+                search_result: str = self.search_agent.run(character.name + ' ' + user_input)
+                search_context = 'Search input: ' + user_input + '\n' + 'Search result: ' + search_result
+                logger.info(f'Search result: {search_context}')
+                # Append to context
+                context += '\n' + search_context
+                # Add back to context store
+                self.db.add_texts([search_context], [{
```

Comment:
qq, why does it need to add in db? since if search enabled, it will always search internet

Reply:
Umm yeah I was originally planning to have the smart mode to automatically decide if need to search web. In that case, it makes sense to add in db for future reference. 
But I found it is not easy job so I want to reserve it for future work. I can remove this part if you think it should not be included in this simple mode.